### DISCUSSION TITLE
Where is the UI and contract interface in the dapp-card-store template?

### DISCUSSION THREAD

**Discussion by sachinkadyan7** [Agoric/agoric-sdk#7634] [Q&A] [answered]
Hey all,

I am trying to implement a smart contract to mint NFTs in my app. I am using the `dapp-card-store` [template](https://github.com/Agoric/dapp-card-store) to understand and take the parts of the contract that will be useful in my app.

For this, I need to know what all parts of the app invoke the contract so that I can rip the contract from the card dapp, and implemented it in my own. Unfortunately the point of interfacing between the contract and the UI in that template is a little bit convoluted, so I need help from the community and from the Agoric team in figuring this part out.

My basic purpose is to take the contract and just transfer it over into my own app.

![Screenshot 2023-05-07 152639](https://user-images.githubusercontent.com/19634735/236698590-efb27b21-0623-46d1-a1ad-7a5b2d05f864.png)

**Comment by sachinkadyan7** [comment]
@dckc 
Hey all,

Any updates on this question? We have quite a bit of time on our end watching the videos but it is still not clear to us where the UI and contract interface is in the dapp-card-store template.

Would really appreciate some help here.

**Reply by dckc** [reply]
Oops! Sorry for the delay in responding. Thanks for tagging me.

I'm not familiar with the code. Here's hoping I find time to look into it. Meanwhile...

@samsiegart @ivanlei do you happen to know off hand? Maybe @michaelfig ?

**Reply by samsiegart** [reply, marked as answer]
The UI uses CapTP to interface with the contract through the ag-solo. For instance, here [publicFacet](https://github.com/Agoric/dapp-card-store/blob/main/ui/src/auction.js#L17) is the public facet of the contract, which the UI uses to get an invitation. Then it [sends that invitation](https://github.com/Agoric/dapp-card-store/blob/main/ui/src/auction.js#L39) to the ag-solo, which will present the offer in the wallet UI, and once approved, send it to zoe to execute.
